File: leetcode/#102.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        queue<TreeNode*> q;
        
        if(root == NULL)
            return {};
        
        vector<int> tmp;
        vector<vector<int>> answer;
        
        q.push(root);
        
        while(!q.empty()){
            int n = q.size();
            
            tmp.clear();
            for(int i = 0; i < n; i++){
                if(q.front()->left)
                    q.push(q.front()->left);
                if(q.front()->right)
                    q.push(q.front()->right);
                
                tmp.push_back(q.front()->val);
                q.pop();
                
            }
            
            answer.push_back(tmp);
        }
        return answer;
    }
};

File: leetcode/#102.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        if root == None:
            return []
        
        ans, q = [], [root]
        
        while len(q) > 0:
            num, tmp = len(q), []
            for _ in range(num):
                cur = q[0]
                q.pop(0)
                tmp.append(cur.val)
                
                if cur.left != None:
                    q.append(cur.left)
                if cur.right != None:
                    q.append(cur.right)
            ans.append(tmp)
            
        return ans